portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java

```diff
@@ -1451,6 +1454,33 @@ public Void call() throws Exception {
 
 		TransactionCommitCallbackUtil.registerCallback(callable);
 
+		// SystemEvent
+
+		DeleteSystemEventActionableDynamicQuery
+			deleteSystemEventActionableDynamicQuery =
+				new DeleteSystemEventActionableDynamicQuery(
+					company.getCompanyId());
+
+		deleteSystemEventActionableDynamicQuery.performActions();
+
+		// ExpandoTable
+
+		DeleteExpandoTableActionableDynamicQuery
+			deleteExpandoTableActionableDynamicQuery =
+				new DeleteExpandoTableActionableDynamicQuery(
+					company.getCompanyId());
+
+		deleteExpandoTableActionableDynamicQuery.performActions();
+
+		// ExpandoColumn
+
+		DeleteExpandoColumnActionableDynamicQuery
+			deleteExpandoColumnActionableDynamicQuery =
+				new DeleteExpandoColumnActionableDynamicQuery(
+					company.getCompanyId());
+
+		deleteExpandoColumnActionableDynamicQuery.performActions();
+
```

Comment:
I'm not sure these should go here.  Generally we want to delete higher-level objects earlier in the sequence (before the company itself is unregistered).

If it's indeed ok to delete these things after the company is unregistered, we could also explore using a ModelListener to remove these items.